### ISSUE TITLE
Support for LDoc/Luadoc

### ISSUE THREAD

**Issue by ghost** [LuaLS/lua.tmbundle#1]
- [ ] @ tags inside of block comments
- [ ] `--` tag prefix as well as `---`
- [ ] redundant `-{1,4}`prefix support in block comment 
- [ ] stretch goal - highlight tagless `---` as LDoc summary/header (likely requiring LSP)

**Comment by sumneko** [comment]
As far as I know, the rules of LDoc have many conflicts with EmmyLua, and I don't understand its detailed rules.
Can you make a PR?

**Comment by ghost** [comment]
I personally think someone should roll their own story of a first-class Lua documentation system.  
This would have syntax and functionality matching JSDoc, and a taglet API to generate documentation data in JSON.

**Comment by Sainan** [comment]
I don't see why you give such a big laundry list, when all that's really needed is classifying the @-thingies as entity.name.tag.documentation, right?

![image](https://user-images.githubusercontent.com/63328889/231689182-dbedfa62-c341-4818-91c7-6cb36d702c0f.png)

This can be achieved by adding the following to `comment.line.double-dash.lua`:
```XML
<key>patterns</key>
<array>
	<dict>
		<key>match</key>
		<string>\@(param|see|comment|usage|return|field|author|set|class|name|pragma|alias|copyright|summary|description|release|license|fixme|todo|warning|raise|charset|within|local|export|private|constructor|static|include|module|script|example|topic|submodule|classmod|file|function|lfunction|table|section|type|annotation|factory)</string>
		<key>name</key>
		<string>entity.name.tag.documentation.lua</string>
	</dict>
</array>
```

**Comment by zm-cttae** [comment]
Fixed in #10, needs closing. :)
